### PR TITLE
Added stop action for chat

### LINKED ISSUE #1266: Add support for "stop" action in SUSI response
**Actual Behaviour**

Currently, new 'stop' action is introduced but the action is not implemented.

**Expected Behaviour**

With `stop` input SUSI should stop anything it's doing at that time. 'stop' action is introduced to enable user to make SUSI stop anything it's doing.
Try to make SUSI respond to “Stop” (e.g. SUSI is narrating a story and if the user says STOP, it should stop narrating the story).

**Steps to reproduce it**

Ask SUSI _stop_, you will get the following actions object:
```
"actions": [{"type": "stop"}],
```
Helpful Links: [skill_link
](https://github.com/fossasia/susi_server/blob/development/conf/susi/en_0001_foundation.txt#L41)

Sample JSON Response:
```
{
		query: "stop",
		count: 1,
 		client_id: "aG9zdF8xNzIuNjguMTQ0LjU3X2Q5NzI1MGMx",
 		query_date: "2018-05-19T22:18:48.641Z",
 		answers:  [
     		 {
         		data:  [
             		 {
                 		0: "stop",
                 		actions:  [
                     		 {
                         		type: "stop"
                     		}
                 		],
                 		token_original: "stop",
                 		token_canonical: "stop",
                 		token_categorized: "end",
                 		timezoneOffset: "-330",
                 		language: "en",
                 		skill_link: "https://github.com/fossasia/susi_server/blob/development/conf/susi/en_0001_foundation.txt",
                 		query: "stop",
                 		skill: "/susi_server/conf/susi/en_0001_foundation.txt"
             		},
             		 {
                 		skill_link: "https://github.com/fossasia/susi_server/blob/development/conf/susi/en_0001_foundation.txt",
                 		query: "stop",
                 		skill: "/susi_server/conf/susi/en_0001_foundation.txt"
             		},
             		 {
                 		skill_link: "https://github.com/fossasia/susi_server/blob/development/conf/susi/en_0001_foundation.txt",
                 		query: "stop",
                 		skill: "/susi_server/conf/susi/en_0001_foundation.txt"
             		},
             		 {
                 		skill_link: "https://github.com/fossasia/susi_server/blob/development/conf/susi/en_0700_ai_play.json",
                 		query: "hey susi",
                 		skill: "/susi_server/conf/susi/en_0700_ai_play.json"
             		}
         		],
         		metadata:  {
             		count: 4
         		},
         		actions:  [
             		 {
                 		type: "stop"
             		}
         		],
         		skills:  [
             		"/susi_server/conf/susi/en_0001_foundation.txt"
         		],
         		persona: { }
     		}
 		],
 		answer_date: "2018-05-19T22:18:48.646Z",
 		answer_time: 5,
 		language: "en",
 		session:  {
     		identity:  {
         		type: "host",
         		name: "172.68.144.57_d97250c1",
         		anonymous: true
     		}
 		}
}
```

**Screenshots of the issue**

N/A

**Would you like to work on the issue?**

No

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1266 

Changes: Whenever susi is dictating a long message, just typing stop will stop susi from further dictating the message and a response "Stopped" will be replied to the user.

Screenshots for the change: 

screenshot_20180525-175901

### CODE REVIEW

**Comment by batbrain7** [comment]
@iamareebjamal @the-dagger @chiragw15 @amitiwary999 @arundhati24 
Please review!

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt [hunk 1]
```diff
@@ -48,7 +48,7 @@ import java.util.*
  * The V in MVP
  * Created by chiragw15 on 9/7/17.
  */
-class ChatActivity: AppCompatActivity(), IChatView {
+class ChatActivity : AppCompatActivity(), IChatView {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: No space should be there before colon

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: OK. But refrain from formatting code in PRs, use separate PR for such stuff

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatActivity.kt]: Sure @iamareebjamal removing this space

File: app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt [hunk 2]
```diff
@@ -76,6 +77,12 @@ class ParseSusiResponseHelper {
                 webSearch = ""
             }
 
+            Constant.STOP -> try {
+                stop = susiResponse.answers[0].actions[1].type
+            } catch (e: Exception) {
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: Why not logging the exception

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: @iamareebjamal it was not done in any other cases previously

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: if this should be done should it not be for all the cases?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: Create an issue to log all caught exceptions. Generic exceptions should not even be caught actually

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: @iamareebjamal created a new issue for that purpose #1314

**Comment by iamareebjamal** [comment]
@batbrain7 Please resolve conflicts

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt [hunk 3]
```diff
@@ -376,9 +376,11 @@ class ChatPresenter(chatActivity: ChatActivity) : IChatPresenter, IChatModel.OnR
                 handler.postDelayed({
                     val psh = ParseSusiResponseHelper()
                     psh.parseSusiResponse(susiResponse, actionNo, utilModel.getString(R.string.error_occurred_try_again))
-                    var setMessage = psh.answer;
+
+                  var setMessage = psh.answer;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt]: Wrong indentation

**Pull request merged**